File: Weather Gambler/app/src/main/java/com/example/cupcake/network/ApiService.java
package com.example.cupcake.network;


import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.Response;

public class ApiService {
    public static String fetchJsonData() {
        OkHttpClient client = new OkHttpClient();
        String url = "https://api.weatherbit.io/v2.0/forecast/daily?city=Pheonix,AZ&key=e932f12ab1cf4ec0b3c81e0f61a506a9"; // Replace with your actual API URL
        Request request = new Request.Builder()
                .url(url)
                .build();

        try (Response response = client.newCall(request).execute()) {
            return response.body().string();
        } catch (Exception e) {
            e.printStackTrace();
            return "{}"; // Return an empty JSON object in case of failure
        }
    }
}

File: Weather Gambler/app/src/main/java/com/example/cupcake/network/apiService.kt
package com.example.cupcake.network

import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext
import okhttp3.OkHttpClient
import okhttp3.Request
import java.io.IOException

object ApiService {
    suspend fun fetchJsonData(): String = withContext(Dispatchers.IO) {
        val client = OkHttpClient()
        val url = "https://api.weatherbit.io/v2.0/forecast/daily?city=Pheonix,AZ&key=YOUR_API_KEY" // Replace YOUR_API_KEY with your actual API key
        val request = Request.Builder()
            .url(url)
            .build()

        try {
            client.newCall(request).execute().use { response ->
                if (!response.isSuccessful) throw IOException("Unexpected code $response")

                response.body?.string() ?: "{}" // Return an empty JSON object in case of failure or null body
            }
        } catch (e: Exception) {
            e.printStackTrace()
            "{}" // Return an empty JSON object in case of failure
        }
    }
}
